### PR TITLE
cairo_enums

### PR DESCRIPTION
**Pull request by ferdousi**
cairo cairo context

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        self.rgba.red = red;
+        self.rgba.green = green;
+        self.rgba.blue = blue;
+        self.rgba.alpha = alpha;
+
+    }
+
+}
+
```
Comment (kneitinger) [inline src/context.rs]: Ensure tests appropriately verify the intended behavior

File: src/context.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * Contributor(s):
+ *	Sara Ferdousi <ferdousi@pdx.edu>
+ *
+ */
+
+use surfaces::ImageSurface;
+use types::Rgba;
+
+
```
Comment (bobbyesh) [inline src/context.rs]: Context needs some user-facing documentation, as it will be auto-generated in the docs.

File: src/context.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+pub struct Context<'a>{
+
+    pub rgba: Rgba,
+    target: &'a ImageSurface,
+
+}
+
+impl<'a> Context<'a>{
+
```
Comment (bobbyesh) [inline src/context.rs]: Missing user-facing documentation for Context::create

File: src/context.rs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    pub fn create(target: &'a ImageSurface )-> Context {
+
+        Context{
+            rgba: Rgba::new(0., 0., 0., 0.),
+            target: target,
+        }
+    }
+
+    pub fn set_source_rgba(&mut self, red: f32, green: f32, blue: f32, alpha: f32){
```
Comment (bobbyesh) [inline src/context.rs]: Missing user-facing documentation for Context::set_source_rgba

File: src/context.rs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+    pub rgba: Rgba,
+    target: &'a ImageSurface,
+
+}
+
+impl<'a> Context<'a>{
+
+    pub fn create(target: &'a ImageSurface )-> Context {
+
```
Comment (bobbyesh) [inline src/context.rs]: I'd vote for eliminating unnecessary whitespace, as per the std lib style...

Check out this function from std for reference:
https://github.com/rust-lang/rust/blob/master/src/libstd/collections/hash/map.rs#L641

**Comment by eSmelser** [comment]
There is one minor merge conflict that needs to be resolved here in lib.rs. I presume we will want to include both, but had a question about the dead code allow declaration. I wasn't sure if we needed to add that line to the line being added from the cairo_enums branch before merging in to develop.

**Comment by eSmelser** [comment]
Reviewed all the code and fixed the Merge conflicts. Everything looks great. Well done @ferdousi !!
